vspec/model/constants.py

```diff
@@ -59,46 +83,79 @@ class StringStyle(Enum, metaclass=EnumMetaWithReverseLookup):
 
 
 class Unit(Enum, metaclass=EnumMetaWithReverseLookup):
-    MILIMETER = "mm"
-    CENTIMETER = "cm"
-    METER = "m"
-    KILOMETER = "km"
-    KILOMETERPERHOUR = "km/h"
-    METERSPERSECONDSQUARED = "m/s^2"
-    LITER = "l"
-    DEGREECELCIUS = "celsius"
-    DEGREE = "degrees"
-    DEGREEPERSECOND = "degrees/s"
-    KILOWATT = "kW"
-    KILOWATTHOURS = "kWh"
-    KILOGRAMM = "kg"
-    VOLT = "V"
-    AMPERE = "A"
-    SECOND = "s"
-    MINUTE = "min"
-    WEEKS = "weeks"
-    MONTHS = "months"
-    UNIXTIMESTAMP = "UNIX Timestamp"
-    PASCAL = "Pa"
-    KILOPASCAL = "kPa"
-    PERCENT = "percent"
-    CUBICMETER = "cm^3"
-    HORSEPOWER = "PS"
-    STARS = "stars"
-    GRAMMPERSECOND = "g/s"
-    GRAMMPERKM = "g/km"
-    KILOWATTHOURSPER100KM = "kWh/100km"
-    LITERPER100KM = "l/100km"
-    LITERPERHOUR = "l/h"
-    MILESPERGALLON = "mpg"
-    POUND = "lbs"
-    NEWTONMETER = "Nm"
-    REVOLUTIONSPERMINUTE = "rpm"
-    INCH = "inch"
-    RATIO = "ratio"
+    """Data Unit Types
+
+    A signal can optionally specify a unit of measurement from the
+    following set. This list composed with definition according to
+    International Units (SI) and few automotive specific units:
+    [Specification](https://www.iso.org/standard/30669.html),
+    [Wikipedia](https://en.wikipedia.org/wiki/International_System_of_Units)
+    """
+    MILIMETER = VSSConstant("mm", "Millimeter", "Distance")
+    CENTIMETER = VSSConstant("cm", "Centimeter", "Distance")
+    METER = VSSConstant("m", "Meter", "Distance")
+    KILOMETER = VSSConstant("km", "Kilometer", "Distance")
+    KILOMETERPERHOUR = VSSConstant("km/h", "Kilometers per hour", "Speed")
+    METERSPERSECONDSQUARED = VSSConstant(
+        "m/s^2",
+        "Acceleration in meters per second squared",
+        "Acceleration"
+    )
+    LITER = VSSConstant("l", "Liter", "Volume")
+    DEGREECELCIUS = VSSConstant("celsius", "Degrees Celsius", "Temperature")
+    DEGREE = VSSConstant("degrees", "Angle in degrees", "Angle")
+    DEGREEPERSECOND = VSSConstant(
+        "degrees/s",
+        "Angular speed in degrees/s",
+        "Angular Speed"
+    )
+    KILOWATT = VSSConstant("kW", "Kilowatt", "Power")
+    KILOWATTHOURS = VSSConstant("kWh", "Kilowatt hours", "Power")
+    KILOGRAMM = VSSConstant("kg", "Kilograms", "Weight")
+    VOLT = VSSConstant("V", "Potential difference in volt", "Electrical")
```

Comment:
Why do we document units?

I mean, If you do want to understand volts, why not look it up: https://en.wikipedia.org/wiki/Volt

Reply:
I can see some use of "Domain", like "Power" or "Volume" to be able to quickly find out which units that VSS (and related tools) support in a certain domain. For example, if I want to specify a signal for the maximum power consumption of the front wipers, I could check for units marked with "Power" and find "kW", "kWh" and "PS" and select one of them. That goes quicker than checking unit one by one, not all persons know in their head what e.g. "PS" and "lbs" are used for.

(But if used for that purpose "Power" is not a good description for "kWh", it would rather be "Energy")

Reply:
Documentation becomes more essential if we have units that might be ambiguous. A typical example could be "stars" below. To be useful I think that it needs some explanation. Do we e.g. refer to a particular star rating (e.g. Euro NCAP crash rating) or could it be any star rating using any scale? If we give some documentation, like "rating, minimum value is 0, maximum value is context/signal dependent" then that information could possible be used by UI designers as a hint on how to present a signal using "star" in a UI.